test/src/unit-filter.cc

```diff
@@ -0,0 +1,321 @@
+          .ok());
+  CHECK(max_window_size0 == max_window_size1);
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test bit shuffle filter deserialization",
+    "[filter][bit-shuffle]") {
+  BitshuffleFilter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test byte shuffle filter deserialization",
+    "[filter][byte-shuffle]") {
+  ByteshuffleFilter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test checksum md5 filter deserialization",
+    "[filter][checksum-md5]") {
+  ChecksumMD5Filter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test checksum sha256 filter deserialization",
+    "[filter][checksum-sha256]") {
+  ChecksumSHA256Filter filter0;
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test encryption aes256gcm filter deserialization",
+    "[filter][encryption-aes256gcm]") {
+  EncryptionKey encryptionkey0;
+  encryptionkey0.set_key(
+      EncryptionType::AES_256_GCM, "abcdefghijklmnopqrstuvwxyz012345", 32);
+  EncryptionAES256GCMFilter filter0(encryptionkey0);
+
+  Buffer buffer;
+  filter0.serialize(&buffer);
+
+  ConstBuffer constbuffer(&buffer);
+  auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+  CHECK(st_filter.ok());
+
+  // Check type
+  CHECK(filter0.type() == filter1.value()->type());
+  buffer.clear();
+}
+
+TEST_CASE(
+    "Filter: Test compression filter deserialization",
+    "[filter][compression]") {
+  SECTION("no level compression") {
+    auto compressortype0 = GENERATE(
+        Compressor::NO_COMPRESSION, Compressor::RLE, Compressor::DOUBLE_DELTA);
+    int compressionlevel0 = 0;
+    CompressionFilter filter0(compressortype0, compressionlevel0);
+    Buffer buffer;
+    filter0.serialize(&buffer);
+
+    ConstBuffer constbuffer(&buffer);
+    auto&& [st_filter, filter1]{Filter::deserialize(&constbuffer)};
+    CHECK(st_filter.ok());
+
+    // Check type
+    CHECK(filter0.type() == filter1.value()->type());
+    buffer.clear();
```

Comment:
check for compression level as well?

Reply:
The compressors in this section do not have a compression level parameter.